tuf/src/database.rs

```diff
@@ -99,21 +100,60 @@ impl<D: DataInterchange> Database<D> {
         })
     }
 
+    /// Create a new [`Database`] struct from a set of metadata that is assumed to be trusted. that
+    /// are used to verify the signed metadata. The signed root metadata must be signed with at
+    /// least a `root_threshold` of the provided root_keys. It is not necessary for the root
+    /// metadata to contain these keys.
+    pub fn from_metadata_with_trusted_keys<'a, I>(
+        metadata: &RawSignedMetadataSet<D>,
+        root_threshold: u32,
+        root_keys: I,
+    ) -> Result<Self>
+    where
+        I: IntoIterator<Item = &'a PublicKey>,
+    {
+        let mut db = if let Some(root) = metadata.root() {
```

Comment:
This reads to me as "make a database if there's root metadata, but don't use the root keys." Why would we want to override the signing keys but use the rest of the root metadata?

Reply:
Carrying over from the other thread, the `root_keys` are only used to establish trust with the root role in `metadata`. Once we've established trust (which happens when we call `Database::from_root_with_trusted_keys`), we'll only trust the keys in the now trusted root role, not the `root_keys`. Does that make sense?

Reply:
Yup, thanks.